### DISCUSSION TITLE
Help With Flashing Firmware On BlueRetro (PS2)

### DISCUSSION THREAD

**Discussion by ultenhiemer** [darthcloud/BlueRetro#901] [Q&A] [answered]
I just got a BlueRetro adapter for my PS2.

It mentioned it was firmware updatable.

I am having problems understanding how to do that properly. As the manual is very confusing.

The manual _(https://www.retroscaler.com/wp-content/uploads/2023/06/Blueretro-PS1PS2-manual.pdf)_ goes over the firmware update on page 24+.

It tells you to download _"NodeMCU-PyFlasher"_ and _"Blueretro firmware"_ from _(https://retroscaler.com/archives/415)_ ... Though that link is just a copy of the home page and has no support option/pages to download these files from them. I had to search and download them separately.

The problem I am at is on page 25. I put in the COM port for my BlueRetro device in the NodeMCU-PyFlasher... But I am completely unsure as to what to properly use for the _"NodeMCU Firmware"._
The manual has _"F:\etroscaler-blueretro-v1.7.1.bin"_ put in that box, but none of the .bin files within the _"v1.9.1_hw2.zip"_ come even close to looking like that one file, and there are many of them.

Which is the proper file to use for flashing the BlueRetro PS2 adapter? The manual is obviously very flawed in it's instructions. Could somebody guide me?

**Comment by ButtonBasher92** [comment, marked as answer]
The manual that comes with the BlueRetro for PS2 in incorrect and should be updated, don't update this way, I messed up my first BlueRetro following the manual, was a whole process to get it working again,

Instead download the latest version from here https://darthcloud.itch.io/blueretro
you will need "V1.9.1_hw1.zip"
unzip it, "BlueRetro_hw1_playstation.bin" is the file you will need
In Chrome go to https://blueretro.io/
plug your Blueretro into your PC with the supplied USB C cable
Click on "BlueRetro OTA FW Update"
Click "Connect BlueRetro"
Select youre BlueRetro from the pop up window and click "Pair"
Choose the file "BlueRetro_hw1_playstation.bin" from the file you just uzipped
And Update

**Reply by djmtech** [reply]
Hi I flashed over usb using the nodeMCU program that is in the manual and now my device doesn't sync. The sync light wont flash. How did you fix yours? Thank you

**Comment by ultenhiemer** [comment]
Thanks, that was a ton of help!
